File: pinpointB/src/main/java/com/nxquar/pinpoint/service/AdminService.java
package com.nxquar.pinpoint.service;


import com.nxquar.pinpoint.DTO.AdminRequest;
import com.nxquar.pinpoint.DTO.AdminResponse;
import com.nxquar.pinpoint.DTO.AuthRequest;
import com.nxquar.pinpoint.DTO.MessageResponse;

import java.util.List;
import java.util.UUID;

public interface AdminService {
    MessageResponse createAdmin(AuthRequest request, String jwt);
    AdminResponse getAdminById(UUID adminId, String jwt);
    List<AdminResponse> getAllAdminsByInstitute(String jwt);
    public AdminResponse getAdminByEmail(String email, String jwt);
    MessageResponse updateAdmin(UUID adminId, AdminRequest request, String jwt);
    MessageResponse deleteAdmin(UUID adminId, String jwt);
}

File: pinpoint/lib/data/datasource/admin/admin_service.dart
import 'package:dio/dio.dart';
import 'package:pinpoint/DTO/auth.dart';
import 'package:pinpoint/model/admin/admin_request.dart';
import 'package:pinpoint/model/admin/admin_response.dart';
import 'package:pinpoint/model/message_response.dart';
import 'package:retrofit/retrofit.dart';
import 'package:pinpoint/resources/constant/string/appString.dart';

part 'admin_service.g.dart';

@RestApi(baseUrl: "${AppString.baseUrl}/api/admins")
abstract class AdminService {
  factory AdminService(Dio dio, {String baseUrl}) = _AdminService;

  @POST("")
  Future<HttpResponse<MessageResponse>> createAdmin(
    @Body() AuthRequest request,
  );
  @GET("/email/{email}")
Future<HttpResponse<AdminResponse>> getAdminByEmail(
  @Path("email") String email,
);


  @GET("/{adminId}")
  Future<HttpResponse<AdminResponse>> getAdminById(
    @Path("adminId") String adminId,
  );

  @GET("/institute/{instituteId}")
  Future<HttpResponse<List<AdminResponse>>> getAdminsByInstitute(
    @Path("instituteId") String instituteId,
  );

  @PUT("/{adminId}")
  Future<HttpResponse<MessageResponse>> updateAdmin(
    @Path("adminId") String adminId,
    @Body() AdminRequest request,
  );

  @DELETE("/{adminId}")
  Future<HttpResponse<MessageResponse>> deleteAdmin(
    @Path("adminId") String adminId,
  );
}
